samples/go/photo-dedup/model/photo.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package model
+
+import (
+	"github.com/attic-labs/noms/go/types"
+	"github.com/attic-labs/noms/samples/go/photo-dedup/dhash"
+)
+
+type Photo interface {
+	ID() ID
+	Dhash() dhash.Hash
+	SetDhash(dhash.Hash)
+	IterSizes(cb func(width int, height int, url string))
+	Marshal() types.Struct
+}
+
+type lazyPhoto struct {
```

Comment:
I don't think you gain much from doing this lazy unmarshalling. The built-in unmarhsal doesn't traverse very far into the tree. And since you typically setup the struct having the shape you want ahead of time, you by definition only unmarshal what you want. I think you should remove this and just use the built-in Unmarshal().